### DISCUSSION TITLE
Set Environment Variable using Powershell

### DISCUSSION THREAD

**Discussion by LazyGeniusMan** [community/community#25713] [Actions]
<p>Could someone help me to set env variable using powershell ? I tried with</p>
<pre><code class="lang-auto">... &gt;&gt; $Env:GITHUB_ENV
</code></pre>
<p>and</p>
<pre><code class="lang-auto">... | Out-File -FilePath $env:GITHUB_ENV -Append
</code></pre>
<p>But it’s not show up when I do <code>${{ env.VAR }}</code></p>
<p><a href="https://github.com/LazyGeniusMan/pyload/actions/runs/1757275021/workflow" rel="noopener nofollow ugc">action run 1</a><br>
<a href="https://github.com/LazyGeniusMan/pyload/actions/runs/1757266840/workflow" rel="noopener nofollow ugc">action run 2</a></p>

**Comment by airtower-luna** [comment]
<p>Variables written to the <code>GITHUB_ENV</code> file aren’t available in the same step. If you add another step after you should be able to use <code>env.VAR</code> there.</p>

**Reply by gbakeman** [reply]
I think this may have been tripping me up - I've spent so much time and several iterations of the same command trying to get an environment variable to stick. I think the problem was, I was echoing back the variable in the same step.

**Comment by LazyGeniusMan** [comment]
<p>I still cant access it</p><aside class="onebox githubactions" data-onebox-src="https://github.com/LazyGeniusMan/pyload/actions/runs/1758998061">
  <header class="source">

      <a href="https://github.com/LazyGeniusMan/pyload/actions/runs/1758998061" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Commit">
      <svg width="24" height="24" class="github-icon github-icon-success" viewbox="0 0 24 24" aria-hidden="true"><path d="M1 12C1 5.925 5.925 1 12 1s11 4.925 11 11-4.925 11-11 11S1 18.075 1 12zm16.28-2.72a.75.75 0 00-1.06-1.06l-5.97 5.97-2.47-2.47a.75.75 0 00-1.06 1.06l3 3a.75.75 0 001.06 0l6.5-6.5z"></path></svg>



  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/LazyGeniusMan/pyload/actions/runs/1758998061" target="_blank" rel="noopener nofollow ugc">Update test_script.yml</a>
    </h4>

      <div class="github-info">
        Test Script <span class="github-run-number">#7</span>
      </div>
  </div>
</div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by LazyGeniusMan** [comment, marked as answer]
<p>Solved it, apparently I dont need to use <code>Write-Host</code> or <code>echo</code> before <code>&gt;&gt;</code><br>
just do<br>
<code>"VAR=TEST" &gt;&gt; $env:GITHUB_ENV</code></p>

**Reply by StephenHodgson** [reply]
Very frustrated that this change from GitHub. It makes such an impact to a lot of my existing PowerShell Scripts. Thanks for this info.

**Reply by Talalgmel** [reply]
U
